main_subnets.tf

```diff
@@ -4,33 +4,36 @@ module "public_subnets" {
   name                    = try(each.value.name, each.key)
   vpc_id                  = aws_vpc.main.id
   vpc_name                = var.name
+  enable_private_subnets  = var.enable_private_subnets
```

Comment:
ideally we don't use a variable here, instead validate if the private subnet var has values, one less variable = simpler code.

Reply:
@hugoalmeida264  am leaving this like this with the intention of creating NAT gateway when there are private subnets (used for conditionals). Changing this introduces some breaking changes which will be handled in the next release as stated in CHANGELOG